javascript/javascript2/week1/script.js

```diff
@@ -0,0 +1,75 @@
+// Find the shortest word
+console.log("---------------------- 1st Task - Find the shortest word ----------------------");
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+
+function findShortestWord(list) {
+  let shortestWordLength = 100,
+    shortestWordIndex;
+  for (var i = 0; i < list.length; i++) {
+    if (shortestWordLength > list[i].length) {
+      shortestWordLength = list[i];
+      shortestWordIndex = i;
+    }
+  }
+  return list[shortestWordIndex];
+}
+console.log("The shortest word in the list is: " + findShortestWord(danishWords)); // returns 'ø'
+
+// Find and count the Danish letters
+console.log("---------------------- 2nd Task - Find Danish letters count ----------------------");
+const danishString1 = "Jeg har en blå bil";
+const danishString2 = "Blå grød med røde bær";
+let danishLettersCount = [0, 0, 0];
+let danishLettersCountTotal = 0;
+
+function findDanishLettersCount(list) {
+  for (var i = 0; i < list.length; i++) {
+    let checkWord = list[i];
+    for (var j = 0; j < checkWord.length; j++) {
+      isDanishLetterFound(checkWord[j]);
+    }
+  }
+  return ("{total: " + danishLettersCountTotal + ", æ: " + danishLettersCount[0] + ", ø: " + danishLettersCount[1] + ", å: " + danishLettersCount[2] + "}");
+}
+
+function isDanishLetterFound(letter) {
+  switch (letter) {
+    case "æ":
+      danishLettersCount[0] = danishLettersCount[0] + 1;
+      ++danishLettersCountTotal;
+      break;
+    case "ø":
+      danishLettersCount[1] = danishLettersCount[1] + 1;
+      ++danishLettersCountTotal;
+      break;
+    case "å":
+      danishLettersCount[2] = danishLettersCount[2] + 1;
+      ++danishLettersCountTotal;
+      break;
+    default:
+  }
+}
+
+console.log(findDanishLettersCount(danishString1)); // returns {total: 1, å: 1}
+console.log(findDanishLettersCount(danishString2)); // returns {total: 4, æ: 1, ø: 2, å: 1}
```

Comment:
Since you are calling your method twice and your 'base' variables are outside of your method, then your counters are not being reset, meaning the second call is not returning  `{total: 4, æ: 1, ø: 2, å: 1}`  but instead `{total: 5, æ: 1, ø: 2, å: 2}`.